gix-blame/src/types.rs

```diff
@@ -81,60 +101,51 @@ impl BlameRanges {
     /// The range should be 1-based inclusive.
     /// If the new range overlaps with or is adjacent to an existing range,
     /// they will be merged into a single range.
-    pub fn add_range(&mut self, new_range: RangeInclusive<u32>) {
-        self.merge_range(new_range);
+    pub fn add_range(&mut self, new_range: RangeInclusive<u32>) -> Result<(), Error> {
+        match self {
+            Self::PartialFile(_) => {
+                let zero_based_range = Self::inclusive_to_zero_based_exclusive(new_range);
+                self.merge_range(zero_based_range)
+            }
+            _ => Err(Error::InvalidOneBasedLineRange),
```

Comment:
Instead of erroring here, I think I prefer turning `self` into `PartialFile` and adding `new_range`. I assume (and hope) that that is what most people would expect.

Reply:
(What I have in mind, seems similar to the logic in `from_one_based_inclusive_ranges`.)